manticore/core/cpu/abstractcpu.py

```diff
@@ -1,64 +1,71 @@
-from ..memory import MemoryException, FileMap, AnonMap
-from ...utils.helpers import issymbolic
-from ...utils.emulate import UnicornEmulator
-from functools import wraps
-from itertools import islice, imap
+
+# FIXME (theo) remove capstone
 import inspect
-import sys
-import types
 import logging
 import StringIO
 
+from .disasm import Capstone
+from abc import abstractmethod
+from functools import wraps
+from itertools import islice, imap
+
+import capstone
+
+from ..smtlib import BitVec, Operators, Constant
+from ..memory import MemoryException
+from ...utils.helpers import issymbolic
+from ...utils.emulate import UnicornEmulator
+
 logger = logging.getLogger("CPU")
 register_logger = logging.getLogger("REGISTERS")
 
 
 
 SANE_SIZES = {8, 16, 32, 64, 80, 128, 256}
-# This encapsulates how to access operands (regs/mem/immediates) for different CPUs
 class Operand(object):
-
+    """This class encapsulates how to access operands (regs/mem/immediates) for
+    different CPUs
+    """
     class MemSpec(object):
         '''
         Auxiliary class wraps capstone operand 'mem' attribute. This will
         return register names instead of Ids
         '''
         def __init__(self, parent):
             self.parent = parent
-        segment = property( lambda self: self.parent._reg_name(self.parent.op.mem.segment) )
-        base = property( lambda self: self.parent._reg_name(self.parent.op.mem.base) )
-        index = property( lambda self: self.parent._reg_name(self.parent.op.mem.index) )
-        scale = property( lambda self: self.parent.op.mem.scale )
-        disp = property( lambda self: self.parent.op.mem.disp )
+        segment = property(lambda self: self.parent._reg_name(self.parent.op.mem.segment))
+        base = property(lambda self: self.parent._reg_name(self.parent.op.mem.base))
+        index = property(lambda self: self.parent._reg_name(self.parent.op.mem.index))
+        scale = property(lambda self: self.parent.op.mem.scale)
+        disp = property(lambda self: self.parent.op.mem.disp)
 
     def __init__(self, cpu, op):
         '''
         This encapsulates the arch-independent way to access instruction
-        operands and immediates based on a capstone operand descriptor. This
-        class knows how to browse a capstone operand and get the details of
-        operand.
+        operands and immediates based on the dissasembler operand descriptor in
+        use. This class knows how to browse an operand and get its details.
 
         It also knows how to access the specific Cpu to get the actual values
         from memory and registers.
 
         :param Cpu cpu: A Cpu instance
-        :param op: A Capstone operand
+        :param op: An operand instance
         :type op: X86Op or ArmOp
         '''
         assert isinstance(cpu, Cpu)
-        assert isinstance(op, (X86Op, ArmOp))
+        # Check against the list of supported operands
+        assert isinstance(op, (capstone.x86.X86Op, capstone.arm.ArmOp))
```

Comment:
That was basically left in place, but once our own Instruction/Operand class is complete, for each disassembler we will retrofit the instructions to fit that, so this will be removed once capstone operands are also ported to the Manticore operands